File: Easy/1-Two-Sum/solutions.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> hasil;
        for (int i = 0; i < nums.size(); i++)
        {
            int selisih = target - nums[i];
            if (hasil.find(selisih) != hasil.end())
            {
                return {i, hasil[selisih]};
            }
            hasil[nums[i]] = i;
        }
        return {};
    }
};

File: Easy/1-Two-Sum/Solutions.ts
function twoSum(nums: number[], target: number): number[] {
    const hasil = {};
    for (let i = 0; i < nums.length; ++i) {
        let selisih = target - nums[i];
        if (selisih in hasil) {
            return [i, hasil[selisih]];
        }
        hasil[nums[i]] = i
    }
    return []
};